### ISSUE TITLE
Feature(CLI): Return file addresses within archive when uploading a directory

### ISSUE THREAD

**Issue by traktion** [maidsafe/autonomi#2434]
I understand that files are now collated within an archive. I assume there are reasons for this (perhaps performance  or cost oriented?), but it is making the upload/download process more difficult in some cases.

However, when an directory is uploaded (as an archive), only the archive address (XOR) is returned. This makes it hard to know how to directly address the files, unless each are uploaded separately. This assumes that directly addressing the files isn't an issue?

Can the CLI return a list of addresses for each item in the archive, along with the archive address itself? Also, can `autonomi file list` do the same?

**Comment by happybeing** [comment]
I agree that this is a problem and have a PR in progress which splits this API into two functions.

One does as now, but is renamed `dir_and_archive_upload()` and the new function `dir_upload()` no longer uploads the Archive but returns it so the caller can inspect the content (files uploaded and xor address). This avoids uploading the Archive unnecessarily, which been be done separately if required.

@traktion how about that?

**Comment by grumbach** [comment]
You can also use the analyse command on the archive address to get the list of files and their addresses. 

Although agree that the Cli should print them out after upload! Added to our backlog!

**Comment by happybeing** [comment]
dweb also has subcommands for inspecting Archives and the other data types.

**Comment by mickvandijke** [comment]
Closed by #2997
